Create srctl tool readme documentation. (#190)
kubernetes/sig-security: sig-security-tooling/srctl/README.md

## Who this tool is for: The SRC

The Security Response Committee (SRC) is responsible for triaging and handling the security issues for Kubernetes. A committee has the license to do work in private, the SRC is the confidential side of sig-security. The primary goal of this group and their process is to reduce the total time users are vulnerable to publicly known exploits.

As part of the above responsibility, the SRC is also responsible for organizing the entire project security response including internal communication and external disclosure but will need help from relevant developers and release managers to successfully run this process.

So the key responsibilities include (with a bit of overlap)
1. security incident response for k8s project infrastructure
2. the identification, triage, disclosure, and fixing of vulnerabilities

> 3. communication and coordination with relevant parties at the appropriate times

Reviewer: I don't think this is the place to document the SRC, I would expect there is some documentation elsewhere that should do the trick that you could link to.

- For Kubernetes core components that are part of a Kubernetes release, provide the CVE feed yaml to the release team, [https://github.com/kubernetes/sig-release/blob/master/release-engineering/role-handbooks/branch-manager.md#announcing-security-fixes](https://github.com/kubernetes/sig-release/blob/master/release-engineering/role-handbooks/branch-manager.md#announcing-security-fixes)
- After public disclosure, [populate CVE details as soon as possible](https://github.com/kubernetes/committee-security-response/blob/main/cna-handbook.md#populate-cve-details-after-public-disclosure)

For more details see:
- https://github.com/kubernetes/committee-security-response/blob/main/security-release-process.md#fix-disclosure-process

### What the problem was.

Vulnerability disclosure needed to be consistently reproduced (no copy paste errors) many times across many communication channels (so reproduce according to many different templates). Additionally, there was need for a more machine friendly OSV feed and SRC is busy enough as can be seen.

> Reducing the report creation workflow to a single writeup process paired with easy automated export processes for the relevant target formats was best to save SRC work, reduce error, and improve usability for the security ecosystem (stable machine-readability). This also makes additional potential channels and formats in the future more approachable.

Reviewer: I feel like this is maybe too much to explain the full SRC disclosure workflow, this could be a short paragraph explaining their pain and how this tool tried to fix it.
```suggestion
Maybe a small paragraph at the end of `## Who this tool is for: The SRC` part?
```

## Command Usage

### CLI Options

> 

Reviewer: your section is empty

Provide a prompt for choice of work to do.

```
CVE-2026-12354
(0) Summary:
(1) CVSS:
(2) Description:
(3) Am I vulnerable?:
(4) Affected Versions:
(5) How to upgrade?:
(6) How to mitigate?:
(7) Detection:
(8) Additional Details:
(9) Acknowledgements:
(a) GitHub Issue:
(b) Fix Lead:

Status: could not find file for CVE-2026-12354, starting from fresh state
Enter or (0-9,a-b) to edit, (s)ave, e(x)port, (q)uit:
```

Notes regarding the file name provided as an argument:
- must match the following regex expression: `^CVE-\d{4}-\d{4,}$`
- is valid with or without the ext which is trimmed off
- if a file with the given valid name
	- does exist in the current directory and the content is fine: the tool will act on the given file
	- does not exist or can't be read by the tool: the tool will create a file with that name and act on that file

Notes on potential problems
- incorrect file name error msg: `invalid CVE name`
- invalid file provided: `error: failed to restore state from file {file name given}: {parsing error}`
- unable to find editor provided at env-var `$EDITOR` in `$PATH`: `failed to read from editor: failed to run the editor: {execution error}`

> - unable to run editor provided in PATH at `$EDITOR`

Reviewer: tbh I wouldn't put that much detail, this feels like it could be quickly outdated and the error displayed by the tool should be enough to understand what went wrong, it should self document

```suggestion
```

Purpose: For saving changed to the report.
Usage Step: Input from the cli process launched with the tool: `s`
Note:
- The tool generates a tmp file which the user operates on in place of the target file during editing so exiting the cli process without telling the tool to save even if saving was done in text editor is not enough. The reasoning here is editing is done back and forth between editor and tool here to work on different sections in isolation easily.

Purpose: For exporting to any supported format.
Supported Formats:
- github issue: [issue.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/issue.tmpl)
- email: [email.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/email.tmpl)
- slack: [slack.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/slack.tmpl)
Usage Steps:
1. Input from the cli process launched with the tool: `e`
2. Input from the cli process launched with the tool: one of `i` issue `m` mail `s` slack `a` all
Note:
- Saving to a JSON file for session state persistence is possible but not a proper disclosure export.

> - The json OSV ([OSV schema](https://ossf.github.io/osv-schema/)) is automatically built for the GitHub Issue.

Reviewer: Again this feels like too much detail, the tool should just be self explanatory, otherwise let's fix it. Supported format could be interesting though since it's the output.

```suggestion
Supported Formats:
- github issue: [issue.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/issue.tmpl)
- email: [email.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/email.tmpl)
- slack: [slack.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/slack.tmpl)
```

### Usage Example

The first example of `srctl` in action was this CVE on which Tabitha used the new tool: https://github.com/kubernetes/kubernetes/issues/136680.

## Implementation

[tooling: add new CLI for CVE publication by SRC members #171](https://github.com/kubernetes/sig-security/pull/171)

> [`srctl` source directory](https://github.com/kubernetes/sig-security/tree/main/sig-security-tooling/srctl)

Reviewer: It's the README I think we should be able to find the sources since there are in the folder we are
```suggestion
```

> # srctl: CVE Report Generation CLI-Tool for the Security Response Committee (SRC)

Reviewer: Maybe a bit shorter?
```suggestion
# srctl: CVE report CLI for the SRC
```

> ## What `srctl` provides.

Reviewer: ```suggestion
```

> The `srctl` tool encodes the standard SRC workflow for vulnerability disclosure and announcement across multiple communication channels into one tool that can handle consistently export to multiple formats (github issue, email, slack). Having this tool also makes adding additional potential channels and formats in the future more approachable. 

Reviewer 2: If this is for SRC, why is it in the SIG Security repo and not an SRC owned repo?

Reviewer 3: SIG Security Tooling is writing it; SRC does not have any code repos or maintain any tools itself at this time.

> The `srctl` tool encodes the standard SRC workflow for vulnerability disclosure and announcement across multiple communication channels into one tool that can handle consistently export to multiple formats (github issue, email, slack). Having this tool also makes adding additional potential channels and formats in the future more approachable. 

Reviewer: Since I removed the full name for the SRC in the title might be good to detail the full name before using the acronym here

- [Kubernetes Security Release Process and Security Committee documentation.](https://github.com/kubernetes/committee-security-response/tree/main)
- [A Bug’s-Eye View: Kubernetes SIG Security Explains It... Ian C, Tabitha S, Rory M, Iain S & Mahé T](https://youtu.be/ujnqeWQZ17w?t=710) from link start time stamp 11:50-21:15
- [SRC Fix Disclosure Process](https://github.com/kubernetes/committee-security-response/blob/main/security-release-process.md#fix-disclosure-process)

Example Usage:
- https://github.com/kubernetes/kubernetes/issues/136680

### Templates & OSV Schema

- Github Issue: [issue.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/issue.tmpl)
- Email: [email.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/email.tmpl)
- Slack: [slack.tmpl src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/slack.tmpl)
- JSON OSV ([OSV schema](https://ossf.github.io/osv-schema/)) is automatically built for the GitHub Issue

## Installation

```
git clone https://github.com/kubernetes/sig-security.git
cd sig-security/sig-security-tooling/srctl
go build
```

> `go install` is not used because it won't work. There is a mismatch of the repository the package lives in on GitHub and the import path recognized from the package name in the `go.mod` file.

Reviewer: off topic for this PR maybe but should we just fix this so that it works?

Author: Not sure how this fix would work considering the nature of each tool being its own fully separate entity in the repo and there ebing potentially go tools that are fully separate like this. I can try to think on it a bit. The other possibility here is shorten to one cmd with git clone and then a Makefile cmd that way everything can be build easily and consistently from the repo root.

Reviewer: this should do the trick https://github.com/kubernetes/sig-security/pull/199

# srctl: CVE report CLI for the SRC

> The `srctl` tool encodes the standard SRC (Security Response Committee) workflow
> for vulnerability disclosure and announcement across multiple communication
> channels into one tool that can handle consistently export to multiple formats
> (github issue, email, slack). Having this tool also makes adding additional
> potential channels and formats in the future more approachable. 

Reviewer 4: ```suggestion
The `srctl` tool encodes the standard SRC (Security Response Committee) workflow
for vulnerability disclosure and announcement across multiple communication
channels into one tool that can consistently export to multiple formats
(github issue, email, slack). Having this tool also makes enabling additional
potential channels and formats in the future more approachable. 
```

Repetitive error prone work of the copy and paste variety is handled by the
tool. SRC members are able to automatically generate a structured json output of
their report in OSV format improving machine readability of vulnerability

> disclosure.

Reviewer 4: ```suggestion
Repetitive error prone work of the "copy and paste" variety is handled by the
tool. SRC members are able to automatically generate a structured `json` output of
their report in `OSV` format improving machine readability of vulnerability
disclosure.
```

> For more details about the SRC who this is for and the workflow behind this see:

Reviewer 4: ```suggestion
For more details about the K8s SRC and the workflow behind this see:
```

  Rory M, Iain S & Mahé T](https://youtu.be/ujnqeWQZ17w?t=710) from link start
  time stamp 11:50-21:15
- [SRC Fix Disclosure
  Process](https://github.com/kubernetes/committee-security-response/blob/main/security-release-process.md#fix-disclosure-process)

Example Usage:
- https://github.com/kubernetes/kubernetes/issues/136680

### Templates & OSV Schema

- Github Issue: [issue.tmpl
  src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/issue.tmpl)
- Email: [email.tmpl
  src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/email.tmpl)
- Slack: [slack.tmpl
  src](https://github.com/kubernetes/sig-security/blob/main/sig-security-tooling/srctl/state/slack.tmpl)
- JSON OSV ([OSV schema](https://ossf.github.io/osv-schema/)) is automatically
  built for the GitHub Issue

## Installation

```
git clone https://github.com/kubernetes/sig-security.git
cd sig-security/sig-security-tooling/srctl
go build
```

`go install` is not used because it won't work. There is a mismatch of the
repository the package lives in on GitHub and the import path recognized from

> the package name in the `go.mod` file.

Reviewer: https://github.com/kubernetes/sig-security/pull/199 go install should work now!! 🥳

Reviewer 4: @Daniel-Giszpenc can you reword this after verifying `go install` works?
